pkg/controller/deadmanssnitch/deadmanssnitch_controller.go

```diff
@@ -292,6 +336,15 @@ func deleteDMS(r *ReconcileDeadMansSnitch, request reconcile.Request, instance *
 		reqLogger.Info("Deleted the DMS from api.deadmanssnitch.com", "Namespace", request.Namespace, "Name", request.Name)
 	}
 
+	// Delete the secret
+	reqLogger.Info("Deleting DMS secret", "Namespace", request.Namespace, "Name", request.Name)
+	err = utils.DeleteSecret(request.Name+config.SecretPostfix, request.Namespace, r.client, reqLogger)
```

Comment:
Should delete the `Secret` after `SyncSet` simply based on the SS referencing the secret.

Reply:
Changed the order